library/common/event/provisional_dispatcher.h

```diff
@@ -69,6 +74,7 @@ class ProvisionalDispatcher : public ScopeTracker {
   std::list<Event::PostCb> init_queue_ GUARDED_BY(state_lock_);
   Event::Dispatcher* event_dispatcher_{};
   Thread::ThreadSynchronizer synchronizer_;
+  bool terminated_ GUARDED_BY(state_lock_){};
```

Comment:
Doesn't matter a ton but if you made this atomic it could be checked outside of the lock

Reply:
Considering that this only needs to be checked in places where we've already acquired the lock, it seems we should leverage that.